src/Botan.php

```diff
@@ -20,152 +23,195 @@
+                            break;
+                        }
                     }
                 }
             }
-        } elseif (isset($obj['edited_message'])) {
-            $data       = $obj['edited_message'];
-            $event_name = 'Edited Message';
-        } elseif (isset($obj['channel_post'])) {
-            $data       = $obj['channel_post'];
-            $event_name = 'Channel Message';
-        } elseif (isset($obj['edited_channel_post'])) {
-            $data       = $obj['edited_channel_post'];
-            $event_name = 'Edited Channel Message';
-        } elseif (isset($obj['inline_query'])) {
-            $data       = $obj['inline_query'];
-            $event_name = 'Inline Query';
-        } elseif (isset($obj['chosen_inline_result'])) {
-            $data       = $obj['chosen_inline_result'];
-            $event_name = 'Chosen Inline Result';
-        } elseif (isset($obj['callback_query'])) {
-            $data       = $obj['callback_query'];
-            $event_name = 'Callback Query';
         }
 
         if (empty($event_name)) {
+            TelegramLog::error("Botan.io stats report failed, no suitable update object found!");
             return false;
         }
 
-        $uid     = $data['from']['id'];
-        $request = str_replace(
-            ['#TOKEN', '#UID', '#NAME'],
-            [self::$token, $uid, urlencode($event_name)],
-            self::$track_url
-        );
-
-        $options = [
-            'http' => [
-                'header'        => 'Content-Type: application/json',
-                'method'        => 'POST',
-                'content'       => json_encode($data),
-                'ignore_errors' => true,
-            ],
-        ];
+        // In case there is no from field assign id = 0
+        if (isset($data['from']['id'])) {
+            $uid = $data['from']['id'];
+        } else {
+            $uid = 0;
+        }
 
-        $context      = stream_context_create($options);
-        $response     = @file_get_contents($request, false, $context);
-        $responseData = json_decode($response, true);
+        try {
+            $response = self::$client->post(
+                str_replace(
+                    ['#TOKEN', '#UID', '#NAME'],
+                    [self::$token, $uid, urlencode($event_name)],
+                    '/track?token=#TOKEN&uid=#UID&name=#NAME'
+                ),
+                [
+                    'headers' => [
+                        'Content-Type' => 'application/json'
+                    ],
+                    'json' => $data
+                ]
+            );
 
-        if ($responseData['status'] !== 'accepted') {
-            TelegramLog::debug('Botan.io API replied with error: ' . $response);
-        }
+            $response = (string) $response->getBody();
+        } catch (RequestException $e) {
+            $response = ($e->getResponse()) ? (string) $e->getResponse()->getBody() : '';
```

Comment:
Looking at the Guzzle `RequestException` class, it seems the response always gets set.

Have you run into a situation where this wasn't the case?

Also, I think variable-name-wise, it would be better not to overwrite `$response` with the response body, as that would actually be `$response_body`.
Maybe call and check the `->getBody()` in the `finally` clause? Also to remove the duplication above.

*(same applies to the other similar code further down)*